src/runtime/tests/manifest-parser-test.ts

```diff
@@ -785,6 +785,48 @@ describe('manifest parser', () => {
         bar: writes Bar {y: Number} = new Bar {y: foo.x}
       `);
     });
+    it('parses from expression', () => {
+      parse(`
+      particle Converter
+        foo: reads Foo {x: Number}
+        bar: writes Bar {y: Number} = from p in foo.x
+      `);
+    });
+    it('parses from expression with nested source', () => {
+      parse(`
+      particle Converter
+        foo: reads Foo {x: Number}
+        bar: writes Bar {y: Number} = from p in (from q in foo.x)
```

Comment:
`from q in foo.x` would not be valid here, as you need to select something from it, e.g. `from q in foo.x select q` (someday we should add the sugar of `foo.x.q`.

Reply:
Is it? I always thought of 'select' as map(), but I can change the grammar to force it

Reply:
I restricted it for now, we can relax later